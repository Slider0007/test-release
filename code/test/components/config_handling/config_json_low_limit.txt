Modify JSON elements accordingly to test lower limits and stringfy as compact version: https://jsonformatter.org/json-stringify-online

{
  "config": {
    "version": 3,
    "lastmodified": ""
  },
  "operationmode": {
    "opmode": -2,
    "automaticprocessinterval": "-1.0",
    "usedemoimages": false
  },
  "takeimage": {
    "flashlight": {
      "flashtime": -100,
      "flashintensity": -1
    },
    "camera": {
      "camerafrequency": -20,
      "imagequality": -12,
      "imagesize": "VGA",
      "brightness": -3,
      "contrast": -3,
      "saturation": -3,
      "sharpness": -5,
      "exposurecontrolmode": -1,
      "autoexposurelevel": -3,
      "manualexposurevalue": -1,
      "gaincontrolmode": -1,
      "manualgainvalue": -1,
      "specialeffect": -1,
      "mirrorimage": false,
      "flipimage": false,
      "zoommode": -1,
      "zoomoffsetx": -1,
      "zoomoffsety": -1
    },
    "debug": {
      "saverawimages": false,
      "rawimageslocation": "/log/source",
      "rawimagesretention": -3
    }
  },
  "imagealignment": {
    "alignmentalgo": -1,
    "searchfield": {
      "x": -20,
      "y": -20
    },
    "imagerotation": "-189.9",
    "flipimagesize": false,
    "marker": [
      {
        "x": 0,
        "y": 0
      },
      {
        "x": -1,
        "y": -1
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "numbersequences": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main"
      }
    ]
  },
  "digit": {
    "enabled": true,
    "model": "dig-class100_0168_s2_q.tflite",
    "cnngoodthreshold": "-0.10",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/digit",
      "roiimagesretention": -3
    }
  },
  "analog": {
    "enabled": true,
    "model": "ana-class100_0171_s1_q.tflite",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/analog",
      "roiimagesretention": -3
    }
  },
  "postprocessing": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "decimalshift": -10,
        "analogdigitsyncvalue": "5.9",
        "extendedresolution": true,
        "ignoreleadingnan": false,
        "checkdigitincreaseconsistency": false,
        "maxratechecktype": -1,
        "maxrate": "0.000",
        "allownegativerate": false,
        "usefallbackvalue": true,
        "fallbackvalueagestartup": -1
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "mqtt": {
    "enabled": false,
    "uri": "",
    "maintopic": "watermeter",
    "clientid": "watermeter",
    "authmode": -1,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "processdatanotation": -1,
    "retainprocessdata": false,
    "homeassistant": {
      "discoveryenabled": false,
      "discoveryprefix": "homeassistant",
      "statustopic": "homeassistant/status",
      "metertype": -1,
      "retaindiscovery": false
    }
  },
  "influxdbv1": {
    "enabled": false,
    "uri": "",
    "database": "",
    "authmode": -1,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "influxdbv2": {
    "enabled": false,
    "uri": "",
    "bucket": "",
    "organization": "",
    "authmode": -1,
    "token": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "gpio": {
    "customizationenabled": false,
    "gpiopin": [
      {
        "gpionumber": 1,
        "gpiousage": "restricted: uart0-tx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": -1,
        "pwmfrequency": -1,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": -1,
          "quantity": 0,
          "colorredchannel": -1,
          "colorgreenchannel": -1,
          "colorbluechannel": -1
        },
        "intensitycorrectionfactor": -1
      },
      {
        "gpionumber": 3,
        "gpiousage": "restricted: uart0-rx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 4,
        "gpiousage": "flashlight-pwm",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "flashlight-default",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 12,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 13,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      }
    ]
  },
  "log": {
    "debug": {
      "loglevel": -1,
      "logfilesretention": -1,
      "debugfilesretention": -1
    },
    "data": {
      "enabled": false,
      "datafilesretention": -1
    }
  },
  "network": {
    "wlan": {
      "ssid": "",
      "password": "",
      "hostname": "watermeter",
      "ipv4": {
        "networkconfig": -1,
        "ipaddress": "",
        "subnetmask": "",
        "gatewayaddress": "",
        "dnsserver": ""
      },
      "wlanroaming": {
        "enabled": false,
        "rssithreshold": -101
      }
    },
    "time": {
      "timezone": "CET-1CEST,M3.5.0,M10.5.0/3",
      "ntp": {
        "timesyncenabled": true,
        "timeserver": "",
        "processstartinterlock": true
      }
    }
  },
  "system": {
    "cpufrequency": 159
  },
  "webui": {
    "autorefresh": {
      "overviewpage": {
        "enabled": true,
        "refreshtime": -1
      },
      "datagraphpage": {
        "enabled": false,
        "refreshtime": -1
      }
    }
  }
}