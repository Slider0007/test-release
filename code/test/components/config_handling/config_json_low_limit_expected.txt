Modify JSON elements accordingly to test lower limits and stringfy as compact version: https://jsonformatter.org/json-stringify-online

{
  "config": {
    "version": 3,
    "lastmodified": ""
  },
  "operationmode": {
    "opmode": -1,
    "automaticprocessinterval": "0.1",
    "usedemoimages": false
  },
  "takeimage": {
    "flashlight": {
      "flashtime": 100,
      "flashintensity": 0
    },
    "camera": {
      "camerafrequency": 5,
      "imagequality": 8,
      "imagesize": "VGA",
      "brightness": -2,
      "contrast": -2,
      "saturation": -2,
      "sharpness": -4,
      "exposurecontrolmode": 0,
      "autoexposurelevel": -2,
      "manualexposurevalue": 0,
      "gaincontrolmode": 0,
      "manualgainvalue": 0,
      "specialeffect": 0,
      "mirrorimage": false,
      "flipimage": false,
      "zoommode": 0,
      "zoomoffsetx": 0,
      "zoomoffsety": 0
    },
    "debug": {
      "saverawimages": false,
      "rawimageslocation": "/log/source",
      "rawimagesretention": 0
    }
  },
  "imagealignment": {
    "alignmentalgo": 0,
    "searchfield": {
      "x": 1,
      "y": 1
    },
    "imagerotation": "-180.0",
    "flipimagesize": false,
    "marker": [
      {
        "x": 1,
        "y": 1
      },
      {
        "x": 1,
        "y": 1
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "numbersequences": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main"
      }
    ]
  },
  "digit": {
    "enabled": true,
    "model": "dig-class100_0168_s2_q.tflite",
    "cnngoodthreshold": "0.00",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/digit",
      "roiimagesretention": 0
    }
  },
  "analog": {
    "enabled": true,
    "model": "ana-class100_0171_s1_q.tflite",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/analog",
      "roiimagesretention": 0
    }
  },
  "postprocessing": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "decimalshift": -9,
        "analogdigitsyncvalue": "6.0",
        "extendedresolution": true,
        "ignoreleadingnan": false,
        "checkdigitincreaseconsistency": false,
        "maxratechecktype": 0,
        "maxrate": "0.001",
        "allownegativerate": false,
        "usefallbackvalue": true,
        "fallbackvalueagestartup": 0
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "mqtt": {
    "enabled": false,
    "uri": "",
    "maintopic": "watermeter",
    "clientid": "watermeter",
    "authmode": 0,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "processdatanotation": 0,
    "retainprocessdata": false,
    "homeassistant": {
      "discoveryenabled": false,
      "discoveryprefix": "homeassistant",
      "statustopic": "homeassistant/status",
      "metertype": 0,
      "retaindiscovery": false
    }
  },
  "influxdbv1": {
    "enabled": false,
    "uri": "",
    "database": "",
    "authmode": 0,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "influxdbv2": {
    "enabled": false,
    "uri": "",
    "bucket": "",
    "organization": "",
    "authmode": 0,
    "token": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "gpio": {
    "customizationenabled": false,
    "gpiopin": [
      {
        "gpionumber": 1,
        "gpiousage": "restricted: uart0-tx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 0,
        "pwmfrequency": 5,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 0,
          "colorgreenchannel": 0,
          "colorbluechannel": 0
        },
        "intensitycorrectionfactor": 1
      },
      {
        "gpionumber": 3,
        "gpiousage": "restricted: uart0-rx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 4,
        "gpiousage": "flashlight-pwm",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "flashlight-default",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 12,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 13,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      }
    ]
  },
  "log": {
    "debug": {
      "loglevel": 1,
      "logfilesretention": 0,
      "debugfilesretention": 0
    },
    "data": {
      "enabled": false,
      "datafilesretention": 0
    }
  },
  "network": {
    "wlan": {
      "ssid": "",
      "password": "",
      "hostname": "watermeter",
      "ipv4": {
        "networkconfig": 0,
        "ipaddress": "",
        "subnetmask": "",
        "gatewayaddress": "",
        "dnsserver": ""
      },
      "wlanroaming": {
        "enabled": false,
        "rssithreshold": -100
      }
    },
    "time": {
      "timezone": "CET-1CEST,M3.5.0,M10.5.0/3",
      "ntp": {
        "timesyncenabled": true,
        "timeserver": "",
        "processstartinterlock": true
      }
    }
  },
  "system": {
    "cpufrequency": 160
  },
  "webui": {
    "autorefresh": {
      "overviewpage": {
        "enabled": true,
        "refreshtime": 1
      },
      "datagraphpage": {
        "enabled": false,
        "refreshtime": 1
      }
    }
  }
}